waffle-chai/src/matchers/emit.ts

```diff
@@ -1,16 +1,30 @@
 import {Contract, providers, utils} from 'ethers';
+import {keccak256, toUtf8Bytes} from 'ethers/lib/utils';
 import {callPromise} from '../call-promise';
 import {waitForPendingTransaction} from './misc/transaction';
 import {supportWithArgs} from './withArgs';
 import {supportWithNamedArgs} from './withNamedArgs';
 
 export function supportEmit(Assertion: Chai.AssertionStatic) {
-  const filterLogsWithTopics = (logs: providers.Log[], topic: any, contractAddress: string) =>
+  const filterLogsWithTopics = (logs: providers.Log[], topic: any, contractAddress?: string) =>
     logs.filter((log) => log.topics.includes(topic))
-      .filter((log) => log.address && log.address.toLowerCase() === contractAddress.toLowerCase());
+      .filter((log) =>
+        log.address &&
+        (contractAddress === undefined || log.address.toLowerCase() === contractAddress.toLowerCase()
+        ));
 
-  Assertion.addMethod('emit', function (this: any, contract: Contract, eventName: string) {
+  const getEventName = (eventSig: string) => {
+    const reg = /(\S+)\((\S*)\)/;
+    const matches = reg.exec(eventSig);
+    if (matches) return matches[1];
+    else throw new Error(`Invalid event signature "${eventSig}"`);
+  };
+
+  Assertion.addMethod('emit', function (this: any, contract: Contract|string, eventName?: string) {
```

Comment:
Don't like this. If necessary consider defining function in a separate place and give it to possible signatures. Also it's better to use self explanatory variable names - e.g. not contract but contractOrEventName